proto/internal/temporal/server/api/persistence/v1/task_queues.proto

```diff
@@ -71,11 +71,56 @@ message CompatibleVersionSet {
     temporal.server.api.clock.v1.HybridLogicalClock became_default_timestamp = 4;
 }
 
+message RampByPercentage {
+    // Acceptable range is [0,100].
+    float ramp_percentage = 1;
+}
+
+message RampByWorkerRatio {
+}
+
+// An internal representation of temporal.api.taskqueue.v1.BuildIdAssignmentRule
+message AssignmentRule {
+    string target_build_id = 1;
+    string hint_filter = 2;
+    // Only terminal Build IDs can have a ramp
+    oneof ramp {
+        RampByPercentage percentage_ramp = 3;
+        RampByWorkerRatio worker_ratio_ramp = 4;
+    }
```

Comment:
I don't think we should duplicate the contents of the public api messages, that just invites a bug where we add a field and forget to copy it. We should define new messages that wrap those and just add the timestamps we need. (This applies to all four messages.)

Reply:
fixed